src/main/java/com/google/opensesame/projects/QueryFilter.java

```diff
@@ -0,0 +1,94 @@
+package com.google.opensesame.projects;
+
+import com.google.opensesame.util.ErrorResponse;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Optional;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * A Datastore filter stored in the form used by Objectify when loading entities from the Datastore.
+ *
+ * <p>See here for how Objectify conducts filtering on a query:
+ * https://www.javadoc.io/static/com.googlecode.objectify/objectify/6.0.6/com/googlecode/objectify/cmd/Query.html#filter(java.lang.String,java.lang.Object)
+ */
+class QueryFilter {
+  public static final String FILTER_QUERY_REGEX = "^[A-Za-z]+ (>|>=|!=|=|<|<=) .+$";
+
+  /**
+   * Parses a QueryFilter from a string and responds with errors to the provided servlet response
+   * object.
+   *
+   * <p>QueryFilter strings should be in the format "fieldName <comparator> value". Examples:
+   *
+   * <blockquote>
+   *
+   * <pre>
+   * "numMentors >= 2"
+   * "numInterestedUsers = 0"
+   * </pre>
+   *
+   * </blockquote>
+   *
+   * @param QueryFilterString The filter query string.
+   * @param response The servlet response to send errors to.
+   * @return Returns the QueryFilter or null if there was an error parsing the string.
+   * @throws IOException
+   */
+  public static QueryFilter fromString(String QueryFilterString, HttpServletResponse response)
+      throws IOException {
+    if (!QueryFilterString.matches(FILTER_QUERY_REGEX)) {
+      ErrorResponse.sendJsonError(
+          response,
+          "Invalid filter query.",
+          HttpServletResponse.SC_BAD_REQUEST,
+          "Unable to query for projects. Please try again later.");
```

Comment:
Is "Please try again later" an error message we want to send here? If it doesn't work now, it's unlikely to work in the future, right? Since this isn't a "server is overloaded" kind of message.

Here and below.

Reply:
Haha that makes sense. Admittedly I wrote that because it sounded professional, but what you said makes sense and I'll remove it.